shows a banner alerting the user if they are using Safari

albireox opened the pull request:
This PR fixes #94, by showing a closable banner when the user is using Safari. The banner looks like

![screen shot 2016-12-10 at 10 43 55](https://cloud.githubusercontent.com/assets/568775/21075434/e7ca4612-bec6-11e6-81f4-9382dd2cf5ab.png)

with the *Learn More* link taking you to the [known issue](https://api.sdss.org/doc/manga/marvin/known_issues.html#known-browser) about Safari. Right now it will show up in all versions of Safari. We could remove it for users using Safari Technology Preview, but since that version is still beta and we are not sure is fully compatible with Marvin, I would leave it. The banner is set to not reappear for a day. This may be a bit annoying. Maybe three or five days is better?

havok2063 commented on python/marvin/web/templates/header.html:
```diff
@@ -1,6 +1,35 @@
+<link rel="stylesheet" type="text/css" href="//cdnjs.cloudflare.com/ajax/libs/cookieconsent2/3.0.3/cookieconsent.min.css" />
+<script src="//cdnjs.cloudflare.com/ajax/libs/cookieconsent2/3.0.3/cookieconsent.min.js"></script>
+<script>
+window.addEventListener("load", function(){
```
I would move this code marvin.js

havok2063 commented:
I see you've finally gotten your hands dirty with the web.  Good!  It looks like this cookieconsent code is primarily used to inform people cookies, which we're using as generic banner code.  That should be ok. At the risk of causing you to swear off the web and curse my name, I would suggest moving this code into the marvin.js file and generalizing it so we could potentially create other banners.  

Also, it looks like detecting browsers and features within browsers has gotten better, and people recommend using Modernizer instead. https://modernizr.com/ If we need to do more browser things,  we might want to switch to this.

havok2063 commented on python/marvin/web/templates/header.html:
```diff
@@ -1,6 +1,35 @@
+<script src="//cdnjs.cloudflare.com/ajax/libs/cookieconsent2/3.0.3/cookieconsent.min.js"></script>
+<script>
+window.addEventListener("load", function(){
+    if (navigator.userAgent.indexOf('Safari') != -1 && navigator.userAgent.indexOf('Chrome') == -1) {
```
alternative to navigator.userAgent, jquery has a browser attribute, accessible via $.browser.   $.browser=='safari' I think should work.  Might clean up the code.

havok2063 commented on python/marvin/web/templates/header.html:
```diff
@@ -1,6 +1,35 @@
+<script>
+window.addEventListener("load", function(){
+    if (navigator.userAgent.indexOf('Safari') != -1 && navigator.userAgent.indexOf('Chrome') == -1) {
+        window.cookieconsent.initialise({
```
can we turn this cookieconsent intialization into a general banner creation, and pass in as variables, the message, dismiss, and href options content?

havok2063 commented on python/marvin/web/templates/header.html:
```diff
@@ -1,6 +1,35 @@
+          },
+          "position": "top",
+          "cookie": {
+              "expiryDays": 1},
```
Does this mean a cookie is created which expires after 1 day?  Does this mean the banner only displays at most once a day?  Is this the behavior we want?  That seems reasonable to me, but just want to double-check.

albireox commented:
Well, there has been a fair amount of swearing, that's for sure. I've moved the code to loayout and to utils.js under a function `marvinBanner`. The Safari banner is now called in `index.html`. I haven't been able to use your `$.browser=='safari'` so I went back to use the same I had.

In `marvinBanner` I have added a few configuration options, but that said, I'm not sure how well this will work as a general banner solution. By default, the library always creates a cookie, so the only way of making the banner work every time you open a certain page is to remove the cookie after it has been created. There are potential issues with domain and path in there. So, if at some point we end up needing a banner solution, there are probably other libraries out there. But for the purposes of this fix, this one works.

For reference, the API documentation of the library is in http://cookieconsent.wpengine.com/documentation/javascript-api/.

havok2063 commented:
The $.browser may be deprecated in jquery, not sure.   Ok, I think this looks reasonable enough.  I've had a look through the docs and yeah, it doesn't seem like this is a general all-purpose banner.  But it should suffice for now.
